src/services/text/TextService.ts

```diff
@@ -15,9 +17,39 @@ export class TextService extends BaseService {
 
   constructor(client: ApiClient, streamingClient: SteamingApiClient) {
     super(client, streamingClient);
-    this.generation = new TextGenerationService(client, streamingClient);
+
+    const generationLimiter = new ConcurrencyLimiter(async () => {
+      const {
+        result: { concurrency },
+      } = await clientErrorWrapper(
+        this._client.GET('/v2/text/generation/limits', {
+          params: { query: { version: '2023-11-22' } },
+        }),
+      );
+      return concurrency;
+    });
+    const embeddingLimiter = new ConcurrencyLimiter(async () => {
+      const {
+        result: { concurrency },
+      } = await clientErrorWrapper(
+        this._client.GET('/v2/text/embeddings/limits', {
+          params: { query: { version: '2023-11-22' } },
+        }),
+      );
+      return concurrency;
+    });
```

Comment:
Why you are creating limiters here? Create it in the concrete service in its constructor.
Same for the `TextGenerationService`.

Reply:
Chat and Generation services share the same limiter.